### PR TITLE
🎉 add barcode and qrcode support

### PR DESCRIPTION
**Pull request by jeankhawand**
image
image
image

### CODE REVIEW

**Comment by ricmars** [comment]
trying to better under the use case - is it a widget in a case view, a field level component (like the signature capture) - is it for self-service on a mobile device? Also we have already a QRCode component in core - should we use it instead?

File: src/components/Pega_Extensions_BarCode/demo.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      selectionProperty: args.selectionProperty,
+      displayValue: args.displayValue,
+      getPConnect: () => {
+        return {
+          getActionsApi: () => {
+            return {
+              openWorkByHandle: () => {
+                /* nothing */
+              },
+              createWork: (className: string) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_BarCode/demo.stories.tsx:94]: should clean up mock for GPConnect to only what is used.

File: src/components/Pega_Extensions_BarCode/demo.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import PegaExtensionsBarCode from './index';
+
+export default {
+  title: 'Widgets/BarCode',
+  argTypes: {
+    heading: {
+      control: 'text'
+    },
+    format: {
+      options: ['CODE128', 'EAN13', 'upc', 'CODE39', 'ITF14', 'MSI', 'pharmacode', 'codabar'],
```
Comment (ricmars) [inline src/components/Pega_Extensions_BarCode/demo.stories.tsx:11]: this should come from your type prop - see other components like displayFormat for Compare table

File: src/components/Pega_Extensions_BarCode/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  format:
+    | 'CODE39'
+    | 'CODE128'
+    | 'EAN13'
+    | 'ITF14'
+    | 'MSI'
+    | 'pharmacode'
+    | 'codabar'
+    | 'upc'
+    | undefined;
```
Comment (ricmars) [inline src/components/Pega_Extensions_BarCode/index.tsx:16]: undefined should not be supported

File: src/components/Pega_Extensions_BarCode/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  selectionProperty: string;
+  displayValue: boolean;
+  getPConnect: any;
+};
+
+/*
+
+*/
+export default function PegaExtensionsBarCode(props: BarCodeProps) {
+  // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
```
Comment (ricmars) [inline src/components/Pega_Extensions_BarCode/index.tsx:26]: no unused vars

File: src/components/Pega_Extensions_BarCode/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <Card>
+      <CardHeader>
+        <Text variant='h2'>{heading}</Text>
+      </CardHeader>
+      <CardContent>
+        <StyledWrapper>
+          <Barcode
+            value={selectionProperty}
+            width={1}
```
Comment (ricmars) [inline src/components/Pega_Extensions_BarCode/index.tsx:37]: should the width be a parameter?

File: src/components/Pega_Extensions_QRCode/demo.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
+import type { StoryObj } from '@storybook/react';
+import PegaExtensionsQRCode from './index';
+
+export default {
+  title: 'Widgets/QRCode',
+  argTypes: {
+    heading: {
```
Comment (ricmars) [inline src/components/Pega_Extensions_QRCode/demo.stories.tsx:7]: not needed

File: src/components/Pega_Extensions_QRCode/demo.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+      heading: args.heading,
+      selectionProperty: args.selectionProperty,
+      getPConnect: () => {
+        return {
+          getActionsApi: () => {
+            return {
+              openWorkByHandle: () => {
+                /* nothing */
+              },
+              createWork: (className: string) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_QRCode/demo.stories.tsx:83]: not needed for this component

File: src/components/Pega_Extensions_QRCode/styles.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import styled, { css } from 'styled-components';
+
+export default styled.div(() => {
+  return css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  img {
+    max-width: 150px;
```
Comment (ricmars) [inline src/components/Pega_Extensions_QRCode/styles.ts:9]: do not use px width - use rem - should height passed as parameter similar to barcode?

File: src/components/Pega_Extensions_QRCode/index.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+  const { heading, selectionProperty, getPConnect } = props;
+  return (
+    <Card>
+      <CardHeader>
+        <Text variant='h2'>{heading}</Text>
+      </CardHeader>
+      <CardContent>
+        <StyledWrapper>
+          <QRCode value={selectionProperty} />
```
Comment (ricmars) [inline src/components/Pega_Extensions_QRCode/index.tsx:23]: missing label

File: src/components/Pega_Extensions_BarCode/index.tsx (hunk 5 above)
Comment (jeankhawand) [reply src/components/Pega_Extensions_BarCode/index.tsx:37]: I reverted it back to the default. This width represents the spacing between bars. Hence, when the value grows in length, there will be a horizontal layout shift. Though, this was the case when in widget mode – I did move it to field which is safe now

Comment (jeankhawand) [reply src/components/Pega_Extensions_BarCode/index.tsx]: It is recommended to stick with 2, using 1 it will make hard to scanner to catch it.

**Pull request merged**